### DISCUSSION TITLE
作者-年 样式中，对两个作者如何使用 和/and 连接（默认对超过1个作者都使用 等/ et al）

### DISCUSSION THREAD

**Discussion by dreistein543** [nju-lug/NJUThesis#189] [Q&A] [answered]
根据一般参考文献要求，对于3个或以上作者才使用 等/et al 进行缩写，对于2个作者应该使用 和/and 连接全部列出。

我查看了 [胡振震的说明文档](https://mirrors.sjtug.sjtu.edu.cn/ctan/macros/latex/contrib/biblatex-contrib/biblatex-gb7714-2015/biblatex-gb7714-2015.pdf#page60) ，尽管其中第14页和26页有提到“用和/and连接”，但是没有具体开启方式。

参考说明文档14页的内容，尝试指定了 gblocal=gb7714-2015 选项，但没有作用。

也尝试手动指定 （导言区加入如下代码）虽然不报错，但是没有起作用
```LaTeX

\usepackage{etoolbox}
\AtEndPreamble{
\DefineBibliographyStrings{english}{
andincite={and},
andincitecn={和},
}
}

```

希望获得大家的帮助

**Comment by hushidong** [comment, marked as answer]
是否启用，是跟模板设置相关的，这个设置主要是引用的标注标签中作者的数量。

需要这么设置：
```
maxcitenames=2,mincitenames=1,uniquelist=false,uniquename=false
```
前两个选项意思是：超过两位作者时，就用一位作者标注该文献，当是两个以下时，则就用全部作者。后面两个选项目的是去掉可能出现的given name的缩写，不这样设置若存在familyname相同时，为区分不同作者会给出givenname缩写。

你在模板里面biblatex设置的时候，加入进去就好了。

**Reply by dreistein543** [reply]
感谢，不过尝试了一下，他只是列出了2个作者，但是是用逗号连接的，并没有使用 和/and

**Reply by dreistein543** [reply]
似乎是需要设置两个names，并结合我提问给的代码才行

**Reply by dreistein543** [reply]
另外直接用 and 前后没有空格，需要用~强制加上空格

**Reply by hushidong** [reply]
是的，前面的选项设置只是针对标注中作者数量的做设置，而本地化字符串或者间隔标点等则是另外设置的，你前面的设置就是本地化字符串，默认是逗号加空格吧，你把它改成and或者和，若要空格也是要加上的。

**Reply by Dongmuyi** [reply]
你好，我在使用\parencite进行引用时，也遇到了同样的问题。在设置完maxcitenames与mincitenames后，列出了2个作者，但作者间用逗号连接。设置了本地化字符串，但是本地化字符串不起作用。希望可以获得你的帮助。本地化字符串代码如下:

> `
\usepackage{etoolbox}
\AtEndPreamble{
\DefineBibliographyStrings{english}{
andincite={and},
andothersincite = {and et al.\isdot},
}
}
`

**Comment by dreistein543** [comment]
> 你好，我在使用\parencite进行引用时，也遇到了同样的问题。在设置完maxcitenames与mincitenames后，列出了2个作者，但作者间用逗号连接。设置了本地化字符串，但是本地化字符串不起作用。希望可以获得你的帮助。本地化字符串代码如下:
> 
> > `\usepackage{etoolbox} \AtEndPreamble{ \DefineBibliographyStrings{english}{ andincite={and}, andothersincite = {and et al.\isdot}, } }`

我当前启用的选择如下，不过我去年毕业了，这是配合 1.2.1 模板使用的，请根据实际情况修改：
```{LaTeX }
\njusetup[bib]{
    style = author-year,
        % 参考文献样式
        % 默认为顺序编码制（numeric）
        % 可选著者-出版年制（author-year）
    option = {
        doi    = false,
        % isbn   = false,
        url    = false,
        % sortcites = true,
        eprint = false,
        % 关闭部分无用文献信息
        % refsection = chapter
        % 将参考文献表置于每章后
        gbnamefmt = lowercase,     % 作者仅首字母大写
        gbalign = right,           % 在 author-year 模式下开启文献序号
        gblanorder = englishahead, % 若要求英文在前 可用 englishahead
        gblabelref = false,        % 作者名取消超链接
        uniquelist = false,
        uniquename = false,        % 启用则取消 L. chen 的名字缩写
        maxcitenames = 2,
        mincitenames = 1,
        maxbibnames = 3,
        minbibnames = 3,
        }
    % 额外的 biblatex 宏包选项
}

\usepackage{etoolbox}
\AtEndPreamble{
    \DefineBibliographyStrings{english}{
        andincite={\protect\space\&\protect\space}, % 这里我使用了 & 作为英文文献双作者分割符，如需使用 and 请修改
        andincitecn={和},
    }
}
```
